Microsoft.Toolkit.Uwp.UI.Controls/MasterDetailsView/MasterDetailsView.cs

```diff
@@ -0,0 +1,241 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using Windows.UI.Composition;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Panel that allows for a Master/Details pattern.
+    /// </summary>
+    public partial class MasterDetailsView : ItemsControl
+    {
+        private ContentPresenter _detailsPresenter;
+        private VisualStateGroup _stateGroup;
+        private VisualState _narrowState;
+        private Frame _frame;
+        private Visual _root;
+        private Compositor _compositor;
+        private Visual _detailsVisual;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MasterDetailsView"/> class.
+        /// </summary>
+        public MasterDetailsView()
+        {
+            DefaultStyleKey = typeof(MasterDetailsView);
+
+            Loaded += OnLoaded;
+            Unloaded += OnUnloaded;
+        }
+
+        /// <summary>
+        /// Invoked whenever application code or internal processes (such as a rebuilding layout pass) call
+        /// ApplyTemplate. In simplest terms, this means the method is called just before a UI element displays
+        /// in your app. Override this method to influence the default post-template logic of a class.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            var detailsPanel = (FrameworkElement)GetTemplateChild("DetailsPanel");
+            _root = ElementCompositionPreview.GetElementVisual(detailsPanel);
+            _compositor = _root.Compositor;
+
+            _detailsPresenter = (ContentPresenter)GetTemplateChild("DetailsPresenter");
+            _detailsPresenter.SizeChanged += OnSizeChanged;
```

Comment:
If I change the template and change the details presenter in some way (change its name, remove it, whatever), this will crash the control. Should have a null check.